File: software/cityjson-info/python/cityjson-info.py
#
#  _____ _ _          __ _____ _____ _____     _     ___     
# |     |_| |_ _ _ __|  |   __|     |   | |___|_|___|  _|___ 
# |   --| |  _| | |  |  |__   |  |  | | | |___| |   |  _| . |
# |_____|_|_| |_  |_____|_____|_____|_|___|   |_|_|_|_| |___|
#             |___|                                          
# 
#   cityjson-info.py
#   Created by Hugo Ledoux on 14/06/2017.
#   Copyright 2017 Hugo Ledoux. All rights reserved.
# 

import json

# fin = open('mycity.json')
fin = open('/Users/hugo/projects/cityjson-new/example-datasets/dummy-values/example2.json')
cm = json.loads(fin.read())

print "CRS:", cm['metadata']['crs']['epsg']

print "There are", len(cm['CityObjects']), "CityObjects"
# list all ids
for id in cm['CityObjects']:
    print "\t", id

print "-"*10

print cm['CityObjects']['2929']

# list all BuildingParts    
for id in cm['CityObjects']:
    if cm['CityObjects'][id]['type'] == 'BuildingPart':
        print "BuildingPart", id


for id in cm['CityObjects']:
    if cm['CityObjects'][id]['type'] == 'TINRelief':
        g = cm['CityObjects'][id]['geometry']
        if 'semantics' in g:
            print "yessss:", g['semantics']
        else:
            print "no semantics"

for id in cm['CityObjects']:
    if cm['CityObjects'][id]['type'] == 'WaterSurface':
        geoms = cm['CityObjects'][id]['geometry']
        for g in geoms:
            if ('semantics' in g) and (g['semantics'] != None):
                print "yessss:", g['semantics']
            else:
                print "no semantics"

# find all Solids
for id in cm['CityObjects']:
    for geom in cm['CityObjects'][id]['geometry']:
        # print geom
        if geom['type'] == "Solid":
            print "Solid LoD:", geom['lod'], id

# vertices
print "number of vertices:", len(cm['vertices'])

# appearance
if 'appearance' in cm:
    print "Appearance:"
    if 'materials' in cm['appearance']:
        print "\t", len(cm['appearance']['materials']), "materials"
    if 'textures' in cm['appearance']:
        print "\t", len(cm['appearance']['textures']), "textures"
    if 'vertex-texture' in cm['appearance']:
        print "\t", len(cm['appearance']['vertex-texture']), "vertex-texture"


for id in cm['CityObjects']:
    geoms = cm['CityObjects'][id]['geometry']
    for g in geoms:
        # print g['texture']
        for each in g['texture']:
            # print each[0]
            if not each[0]:
                print "empty"
            else:
                print each[0]

        break
    break

File: software/cityjson-info/ruby/cityjson-info.rb
#!/usr/bin/env ruby
#  _____ _ _          __ _____ _____ _____     _     ___     
# |     |_| |_ _ _ __|  |   __|     |   | |___|_|___|  _|___ 
# |   --| |  _| | |  |  |__   |  |  | | | |___| |   |  _| . |
# |_____|_|_| |_  |_____|_____|_____|_|___|   |_|_|_|_| |___|
#             |___|                                          

#   cityjson-info
#   Created by Ken Arroyo Ohori on 07/09/2017.
#   Copyright © 2017 Ken Arroyo Ohori. All rights reserved.

require 'json'
require 'set'
require 'pp'

ARGV.length >= 1 ? filename = ARGV[0] : filename = "../../../example-datasets/Rotterdam/Delfshaven/3-20-DELFSHAVEN.json"
input = File.read(filename)
j = JSON.parse(input)

puts "Printing (some) information about:"
puts "  " + filename + "\n"

# CityJSON version
version = j["version"][-3, 3]
puts "CityJSON version: " + version

# CityObjects
puts "===== CityObjects ====="
puts "Total : " + j["CityObjects"].length.to_s
d = Set.new
j["CityObjects"].each do |id, co|
	d.add(co["type"])
end
puts "Types:"
d.each do |each|
	puts "  " + each.to_s
end
d.clear
j["CityObjects"].each do |id, co|
	co["geometry"].each do |g|
		d.add(g["type"])
	end
end
puts "Geometries present:"
d.each do |each|
	puts "  " + each.to_s
end
d.clear

# metadata
puts "===== Metadata ====="
if !j.has_key?("metadata")
	puts "  none"
else
	j["metadata"].each do |id, it|
		if id == "crs" 
			puts "  crs: EPSG:" + j["metadata"]["crs"]["epsg"].to_s
		else
			puts "  " + id
		end
	end
end

# vertices
puts "===== Vertices ====="
puts "Total: " << j["vertices"].length.to_s

# appearance
puts "===== Appearance ====="
if !j.has_key?("appearance")
  puts "  none"
else
  if j["appearance"].has_key?("textures")
    puts "  textures: " + j["appearance"]["textures"].length.to_s
  end
  if (j["appearance"].count("materials") > 0)
    puts "  materials: " + j["appearance"]["materials"].length.to_s
  end
end